src/main/java/cubicchunks/worldgen/gui/component/UIBlockStateButton.java

```diff
@@ -0,0 +1,113 @@
+import cubicchunks.worldgen.gui.DummyWorld;
+import net.malisis.core.client.gui.GuiRenderer;
+import net.malisis.core.client.gui.MalisisGui;
+import net.malisis.core.client.gui.component.UIComponent;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.VertexBuffer;
+import net.minecraft.client.renderer.texture.ITextureObject;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+
+public class UIBlockStateButton extends UIComponent<UIBlockStateButton> {
+
+    private IBlockState iBlockState;
+    private final Consumer<IBlockState> onClick;
+
+    public UIBlockStateButton(MalisisGui gui, IBlockState iBlockState1, Consumer<IBlockState> clickAction) {
+        super(gui);
+        iBlockState = iBlockState1;
+        onClick = clickAction;
+        setTooltip(generateTooltip(iBlockState));
+        setSize(24, 24);
+    }
+
+    private String generateTooltip(IBlockState blockState) {
+        StringBuffer sb = new StringBuffer(128);
+        sb.append(blockState.getBlock().getLocalizedName());
+        for (Entry<IProperty<?>, Comparable<?>> entry : blockState.getProperties().entrySet()) {
+            sb.append(" \n ");
+            sb.append(entry.getKey().getName());
+            sb.append(" = ");
+            sb.append(entry.getValue().toString());
+        }
+        return sb.toString();
+    }
+
+    public void setBlockState(IBlockState iBlockState1) {
+        iBlockState = iBlockState1;
+        setTooltip(generateTooltip(iBlockState));
+    }
+
+    @Override
+    public boolean onClick(int x, int y) {
+        MalisisGui.playSound(SoundEvents.UI_BUTTON_CLICK);
+        onClick.accept(iBlockState);
+        return true;
+    }
+
+    @Override
+    public void drawForeground(GuiRenderer renderer, int mouseX, int mouseY, float partialTick) {
+        if (iBlockState != null) {
+            this.drawBlock(renderer);
+        }
+    }
+
+    private void drawBlock(GuiRenderer renderer) {
+
+        VertexBuffer vertexbuffer = Tessellator.getInstance().getBuffer();
+        Tessellator.getInstance().draw();
+        ITextureObject blockTexture = Minecraft.getMinecraft().getTextureManager().getTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+        GL11.glBindTexture(GL11.GL_TEXTURE_2D, blockTexture.getGlTextureId());
+        VertexFormat format = DefaultVertexFormats.BLOCK;
+        GlStateManager.pushMatrix();
+        GlStateManager.translate((float) this.screenX(), (float) this.screenY() + 16f, 100.0F);
+        GlStateManager.scale(16.0F, -16.0F, 16.0F);
+        GlStateManager.rotate(45f, 1f, 1f, 0f);
```

Comment:
This rotation is slightly wrong...

You are rotating 45 degrees around vector (1, 1, 1) vector, try rotating 45 degrees around (1, 0, 0), then around (0, 1, 0) and then around (0, 0, 1). Also note that changing order will give different result.